### ISSUE TITLE
What would cause a panel function to be called for every kv in ddo per a panel?

### ISSUE THREAD

**Issue by jrounds** [tesseradata/trelliscope#124]
I am being bewildered by something.

Symptom:
every panel in trelliscope looks the same (I think because it is taking the last of many)
Symptom:
panelFn appears to be called once for every kv in a ddo (based on prints) (not once per a panel... once through the ddo for *every single panel*.
Symptom:
the HDFS backed DDO reports "Error in panelContent[[i]] : subscript out of bounds" 

I don't know what to do with these clues.

A cut and paste of some code and output follows, and note the spam of summaries at the end is without touching anything in Trelliscope UI and they repeat for each time a single panel is shifted.

```
> 
> cogFn = function(v){
+ 	y = as.numeric(v$table["total_visits",]) 
+   ny = length(y)
+   y[y==0] = NA 
+  	ret = list()
+  	ret$mean = as.numeric(mean(y, na.rm=TRUE))
+  	ret$sd = as.numeric(sd(y, na.rm=TRUE))
+  	return(ret)
+ 		
+ }
> panelFn = function(v){
+ 	y = as.numeric(v$table["total_visits",]) 
+   ny = length(y)
+   miny = max(1, ny - 3*52)
+ 	data = data.frame(y, v$derived$total_ma_y, v$resid$total_ma_y)
+ 	data = data[miny:ny,]
+ 	zdata = zoo(data, 1:nrow(data))
+ 	#print(zdata)
+ 	print(summary(zdata))
+ 	ret = xyplot(zdata)
+ 	print(ret)
+ 	ret
+ }
> 
> conn = vdbConn("/tmp/test_parallel", autoYes=TRUE)
> test_input = ddo(ddo_mmwr_ma[1:10])
Keys are not known - just retrieving the first 10 key-value pair(s).
Read 10 objects(11.86 MB) in 0.49 seconds
> makeDisplay(test_input, panelFn= panelFn, cogFn=cogFn, name="bob")#, control=ctrl)
* Validating 'panelFn'...
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 356.0   Min.   : 392.8       Min.   :-263.67   
 1st Qu.: 40   1st Qu.: 497.0   1st Qu.: 495.8       1st Qu.: -83.17   
 Median : 79   Median : 606.0   Median : 593.3       Median :   2.75   
 Mean   : 79   Mean   : 671.7   Mean   : 659.1       Mean   :  12.55   
 3rd Qu.:118   3rd Qu.: 776.0   3rd Qu.: 744.6       3rd Qu.:  93.67   
 Max.   :157   Max.   :1586.0   Max.   :1269.8       Max.   : 679.50   
* Testing cognostics function on a subset ... ok
* Precomputed limits not supplied.  Computing axis limits...
* ... skipping this step since both axes are free ...
* Storing display object...
* Plotting thumbnail...
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 356.0   Min.   : 392.8       Min.   :-263.67   
 1st Qu.: 40   1st Qu.: 497.0   1st Qu.: 495.8       1st Qu.: -83.17   
 Median : 79   Median : 606.0   Median : 593.3       Median :   2.75   
 Mean   : 79   Mean   : 671.7   Mean   : 659.1       Mean   :  12.55   
 3rd Qu.:118   3rd Qu.: 776.0   3rd Qu.: 744.6       3rd Qu.:  93.67   
 Max.   :157   Max.   :1586.0   Max.   :1269.8       Max.   : 679.50   
* Updating displayList...
* Display exists... backing up previous to /tmp/test_parallel/displays/common/bob_bak
* Removing previous backup plot directory
> view(conn=conn)
attempting to launch shiny vdb viewer...
press Ctrl+C or Esc to stop viewer

Listening on http://127.0.0.1:8100
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 356.0   Min.   : 392.8       Min.   :-263.67   
 1st Qu.: 40   1st Qu.: 497.0   1st Qu.: 495.8       1st Qu.: -83.17   
 Median : 79   Median : 606.0   Median : 593.3       Median :   2.75   
 Mean   : 79   Mean   : 671.7   Mean   : 659.1       Mean   :  12.55   
 3rd Qu.:118   3rd Qu.: 776.0   3rd Qu.: 744.6       3rd Qu.:  93.67   
 Max.   :157   Max.   :1586.0   Max.   :1269.8       Max.   : 679.50   
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 327.0   Min.   : 367.8       Min.   :-275.083  
 1st Qu.: 40   1st Qu.: 469.0   1st Qu.: 453.2       1st Qu.: -94.750  
 Median : 79   Median : 539.0   Median : 563.2       Median : -14.917  
 Mean   : 79   Mean   : 609.6   Mean   : 605.8       Mean   :   3.714  
 3rd Qu.:118   3rd Qu.: 728.0   3rd Qu.: 723.3       3rd Qu.:  86.333  
 Max.   :157   Max.   :1404.0   Max.   :1095.2       Max.   : 704.583  
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 326.0   Min.   : 349.1       Min.   :-310.00   
 1st Qu.: 40   1st Qu.: 483.0   1st Qu.: 494.9       1st Qu.: -84.33   
 Median : 79   Median : 635.0   Median : 611.5       Median : -35.92   
 Mean   : 79   Mean   : 675.7   Mean   : 665.5       Mean   :  10.20   
 3rd Qu.:118   3rd Qu.: 797.0   3rd Qu.: 790.8       3rd Qu.: 113.75   
 Max.   :157   Max.   :1827.0   Max.   :1312.8       Max.   : 872.83   
     Index           y         v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 2.00   Min.   : 6.167       Min.   :-15.4167  
 1st Qu.: 40   1st Qu.: 9.00   1st Qu.:11.417       1st Qu.: -4.3333  
 Median : 79   Median :15.00   Median :14.083       Median : -0.7500  
 Mean   : 79   Mean   :16.21   Mean   :15.810       Mean   :  0.4002  
 3rd Qu.:118   3rd Qu.:21.00   3rd Qu.:19.500       3rd Qu.:  4.2500  
 Max.   :157   Max.   :44.00   Max.   :29.583       Max.   : 18.4167  
     Index           y        v.derived.total_ma_y v.resid.total_ma_y 
 Min.   :  1   Min.   : 3.0   Min.   : 7.583       Min.   :-19.66667  
 1st Qu.: 40   1st Qu.:14.0   1st Qu.:14.500       1st Qu.: -5.50000  
 Median : 79   Median :20.0   Median :21.750       Median : -0.58333  
 Mean   : 79   Mean   :21.3   Mean   :21.231       Mean   :  0.06794  
 3rd Qu.:118   3rd Qu.:28.0   3rd Qu.:27.000       3rd Qu.:  6.41667  
 Max.   :157   Max.   :45.0   Max.   :37.000       Max.   : 19.91667  
     Index           y     v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   :0   Min.   : NA          Min.   : NA       
 1st Qu.: 40   1st Qu.:0   1st Qu.: NA          1st Qu.: NA       
 Median : 79   Median :0   Median : NA          Median : NA       
 Mean   : 79   Mean   :0   Mean   :NaN          Mean   :NaN       
 3rd Qu.:118   3rd Qu.:0   3rd Qu.: NA          3rd Qu.: NA       
 Max.   :157   Max.   :0   Max.   : NA          Max.   : NA       
                           NA's   :157          NA's   :157       
     Index           y     v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   :0   Min.   : NA          Min.   : NA       
 1st Qu.: 40   1st Qu.:0   1st Qu.: NA          1st Qu.: NA       
 Median : 79   Median :0   Median : NA          Median : NA       
 Mean   : 79   Mean   :0   Mean   :NaN          Mean   :NaN       
 3rd Qu.:118   3rd Qu.:0   3rd Qu.: NA          3rd Qu.: NA       
 Max.   :157   Max.   :0   Max.   : NA          Max.   : NA       
                           NA's   :157          NA's   :157       
     Index           y     v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   :0   Min.   : NA          Min.   : NA       
 1st Qu.: 40   1st Qu.:0   1st Qu.: NA          1st Qu.: NA       
 Median : 79   Median :0   Median : NA          Median : NA       
 Mean   : 79   Mean   :0   Mean   :NaN          Mean   :NaN       
 3rd Qu.:118   3rd Qu.:0   3rd Qu.: NA          3rd Qu.: NA       
 Max.   :157   Max.   :0   Max.   : NA          Max.   : NA       
                           NA's   :157          NA's   :157       
     Index           y     v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   :0   Min.   : NA          Min.   : NA       
 1st Qu.: 40   1st Qu.:0   1st Qu.: NA          1st Qu.: NA       
 Median : 79   Median :0   Median : NA          Median : NA       
 Mean   : 79   Mean   :0   Mean   :NaN          Mean   :NaN       
 3rd Qu.:118   3rd Qu.:0   3rd Qu.: NA          3rd Qu.: NA       
 Max.   :157   Max.   :0   Max.   : NA          Max.   : NA       
                           NA's   :157          NA's   :157       
     Index           y          v.derived.total_ma_y v.resid.total_ma_y
 Min.   :  1   Min.   : 0.000   Min.   :2.333        Min.   :-5.2500   
 1st Qu.: 40   1st Qu.: 2.000   1st Qu.:3.750        1st Qu.:-1.7500   
 Median : 79   Median : 4.000   Median :4.208        Median :-0.5000   
 Mean   : 79   Mean   : 3.904   Mean   :4.449        Mean   :-0.1101   
 3rd Qu.:118   3rd Qu.: 5.000   3rd Qu.:4.917        3rd Qu.: 1.0833   
 Max.   :157   Max.   :21.000   Max.   :7.917        Max.   :14.8333   
                                NA's   :45           NA's   :45    
```

**Comment by hafen** [comment]
Interesting.  Could you check the output of:

```
test_input
```

and

```
length(test_input)
```

**Comment by jrounds** [comment]
> test_input

Distributed data object backed by 'kvMemory' connection

 attribute      | value
----------------+---------------------------------------------------------------------
 size (stored)  | 12.79 MB
 size (object)  | 12.79 MB
 # subsets      | 10

* Other attributes: getKeys()
* Missing attributes: splitSizeDistn

> length(test_input)
[1] 10


I even tried turning it into a list, turning the list into a ddo, and then redoing it and it happens again.

This is in rstudio server.  I am worried this issue is at the intersection of  how rstudio server handles instances,  datadr, trelliscope, and rhipe.  I did my best to reset rstudio and it still happens.

I tried to run trelliscope from the command line R to simplify the problem and got:
```
> view(conn=conn)
attempting to launch shiny vdb viewer...
press Ctrl+C or Esc to stop viewer

Listening on http://127.0.0.1:8100
/usr/bin/xdg-open: line 402: htmlview: command not found
/usr/bin/xdg-open: line 402: firefox: command not found
/usr/bin/xdg-open: line 402: mozilla: command not found
/usr/bin/xdg-open: line 402: netscape: command not found
/usr/bin/xdg-open: line 402: links: command not found
/usr/bin/xdg-open: line 402: lynx: command not found
xdg-open: no method available for opening 'http://127.0.0.1:8100'
```

**Comment by jrounds** [comment]
Q: What would cause a panel function to be called for every kv in ddo per a panel? 
Answer:  Duplicate key hashes will do this (which should never happen).

Diagnosed by Ryan (thanks again).  I will re-issue if I can re-create how to generate dup hashes in a normal workflow.  Otherwise chalking this up to something I did.
